src/hooks/useSelectionLogic.ts

```diff
@@ -165,9 +173,7 @@ export function useSelectionLogic<T extends HTMLElement>({
     (e: Event) => {
       // handle only left button click
       if ((e as MouseEvent).button === 0 && isEnabledRef.current) {
-        // if user clicked on element which is draggable (so it might be start of drag event), ignore it
-        const isDisabled = isSelectionDisabled(e.target as HTMLElement);
-        if (isDisabled) {
+        if (shouldStartSelecting?.(e.target)) {
```

Comment:
Shouldn't we `return` if `shouldStartSelecting` returns `false`?

Reply:
good call. fixed